PyPDF2/_writer.py

```diff
@@ -128,8 +134,9 @@ class (typically :class:`PdfReader<PyPDF2.PdfReader>`).
 
     def __init__(self, fileobj: StrByteType = "") -> None:
         self._header = b"%PDF-1.3"
-        self._objects: List[Optional[PdfObject]] = []  # array of indirect objects
+        self._objects: List[PdfObject] = []  # array of indirect objects
```

Comment:
This is a change that is not related to cloning, right? It seems to be an unrelated improvement you made while writing this PR.

As this PR is rather big and might take a while to review, I would like to move this to a separate PR ("Miscellaneous fixes" or similar)

Reply:
transfered into #1411